packages/lms/src/lib/components/navigation/ProgrammeNav.svelte

```diff
@@ -0,0 +1,105 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { files } from '$lib/stores/files';
+	import { getCurrent } from '$lib/utils/getCurrent';
+	import { browser } from '$app/environment';
+	import { page } from '$app/stores';
+
+	interface Frontmatter {
+		type?: 'Course' | 'Module' | 'Lesson' | 'Institute' | 'Programme';
```

Comment:
Check out the types branch, I'm not 100% sure which is why I haven't opened a PR. 

The idea is to have a basic interface `EntityBase` that just has the most basic metadata like title, type, tags and previous.

Then there will be a "bigger" interface `Entity extends EntityBase` that contains the body text, additional/custom frontmatter, the full path/address of the file and an array of its children. It might require a class, that way we can automatically pull `slug` from `path` by doing some string manipulation.

We just had a power cut but once I am back at my desk I will push updates to the branch.